File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_app_detail/lib/src/services/app_detail_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/app_detail_models.dart';

/// App Detail service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class AppDetailService {
  const AppDetailService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'app-detail';

  AppDetailPageResult<Never> empty() =>
      const AppDetailPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-app-detail/src/services/appDetailService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { AppDetailPageResult } from "../types/appDetailModels.js";

/**
 * App Detail service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface AppDetailService {
  readonly capability: "app-detail";
}

export function createAppDetailService(client: AppstoreAppClient): AppDetailService {
  void client;
  return { capability: "app-detail" };
}

export function emptyAppDetailResult(): AppDetailPageResult<never> {
  return { items: [] };
}
